src/nat/middleware/red_teaming/red_teaming_middleware.py

```
@@ -0,0 +1,338 @@
+            try:
+                if value_type is int or isinstance(original_value, int):
+                    return int(attack_payload)
+                return float(attack_payload)
+            except (ValueError, TypeError) as e:
+                raise ValueError(f"Cannot convert attack payload '{attack_payload}' to {value_type.__name__}") from e
+
+    def _resolve_multiple_field_matches(self, matches):
+        if self._target_field_resolution_strategy == "error":
+            raise ValueError(f"Multiple matches found for target_field: {self._target_field}")
+        elif self._target_field_resolution_strategy == "random":
+            return [random.choice(matches)]
```

Ruff (0.14.8)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)